### PR TITLE
Blog app - controllers specs

### PR DESCRIPTION
**Pull request by AmaliaGM**
## Project requirements

- Add RSpec as a gem to your project.
- Create a Request spec file for all your controllers.
- Make sure that for each action you check:
If response status was correct.
If a correct template was rendered.
If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/user_spec.rb [hunk 1]
```diff
@@ -2,6 +2,29 @@
...
+  end
+
+  describe 'GET /show' do
+    before(:example) { get user_path(User.first.id) }
+
+    it 'returns a 200' do
+      get 'http://localhost:3000/user'
+      expect(response).to have_http_status(:success)
+    end
+
```
Comment (danmainah) [inline spec/requests/user_spec.rb:25]: - [ ] @AmaliaGM great work on your test however it's a requirement to have all test cases for each test, kindly add tests for the template and response.body this is a project requirement,
image

File: spec/requests/user_spec.rb [hunk 2]
```diff
@@ -2,6 +2,29 @@
 
 RSpec.describe 'Users', type: :request do
   describe 'GET /index' do
-    pending "add some examples (or delete) #{__FILE__}"
+    context 'testing GET/index' do
```
Comment (danmainah) [inline spec/requests/user_spec.rb:5]: - [ ] @AmaliaGM great work on your test however it's a requirement to have all test cases for each test, kindly add test for `response.body having correct placeholder text`, this is a project requirement,
image

Comment (danmainah) [inline spec/requests/user_spec.rb:5]: - [ ] @AmaliaGM great work on your test however it's a requirement to have all test cases for each test, kindly add test for `response.body having correct placeholder text`, this is a project requirement,
image

File: spec/requests/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+require_relative '../rails_helper'
+
+RSpec.describe 'Posts', type: :request do
```
Comment (danmainah) [inline spec/requests/post_spec.rb:3]: - [ ]  @AmaliaGM great work on your test however it's a requirement to have all test cases for each test, kindly separate tests for the show and index page, also ensure you have all the tests as shown in the requirements below
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @AmaliaGM , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good design


# Required Changes ♻️

- [ ] Kindly ensure all the tests are running and passing I encountered the errors shown below

image
image


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by AmaliaGM** [comment]
@danmainah hi, how are you? All the test pass in my end

File: spec/requests/user_spec.rb (hunk 1 above)
Comment (AmaliaGM) [reply spec/requests/user_spec.rb:25]: There's a test for the response in users

File: spec/requests/post_spec.rb (hunk 3 above)
Comment (AmaliaGM) [reply spec/requests/post_spec.rb:3]: Hi. They are separated in index and show

File: spec/requests/user_spec.rb (hunk 1 above)
Comment (danmainah) [reply spec/requests/user_spec.rb:25]: @AmaliaGM  yes there is sorry for that

**Comment by danmainah** [comment]
> @danmainah hi, how are you? All the test pass in my end 

@AmaliaGM  let me reset my environment and rerun it again, kindly implement the changes requested and ensure all tests are running

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @AmaliaGM👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- All tests are passed. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AmaliaGM** [comment]
@thecodechaser Thanks a lot! Happy coding!

**Pull request merged**